dvc/repo/artifacts.py

```diff
@@ -0,0 +1,44 @@
+import os
+from typing import TYPE_CHECKING, Any, Dict
+
+from dvc.annotations import Artifact
+from dvc.dvcfile import FileMixin
+from dvc.utils import relpath
+
+if TYPE_CHECKING:
+    from dvc.repo import Repo
+
+
+class ArtifactsFile(FileMixin):
```

Comment:
I'd have preferred not to use `FileMixin`, it's too bloated. But I am fine for now.